pkg/skaffold/build/buildpacks/dependencies.go

```diff
@@ -33,8 +34,27 @@ func GetDependencies(ctx context.Context, workspace string, a *latest.BuildpackA
 		return nil, errors.Wrapf(err, "walking workspace %s", workspace)
 	}
 
-	var dependencies []string
+	// Make sure folders are also listed
+	added := map[string]bool{}
 	for file := range files {
+		for path := file; path != "."; path = filepath.Dir(path) {
```

Comment:
We could speed this up by stopping when a path is already in `added`:
```suggestion
		for path := file; path != "." && !added[path]; path = filepath.Dir(path) {
```

But doesn't this mean that we're now exposing all the contents of those directories too, even if the user specified specific files?  (E.g., `pkg/lib/file.go` will result in dependencies on `pkg/`, `pkg/lib/` too?)

Reply:
I'll move the code to tar.go where it really belongs